source/server/http/admin.cc

```diff
@@ -915,17 +915,21 @@ Http::Code AdminImpl::handlerStatsRecentLookupsEnable(absl::string_view, Http::H
 
 Http::Code AdminImpl::handlerServerInfo(absl::string_view, Http::HeaderMap& headers,
                                         Buffer::Instance& response, AdminStream&) {
-  time_t current_time = time(nullptr);
+  const auto current_time = std::chrono::system_clock::to_time_t(server_.timeSource().systemTime());
```

Comment:
can you use std::time_t rather than `auto`?

https://google.github.io/styleguide/cppguide.html#Type_deduction


Actually what you think of adding TimeSource::systemTimeSeconds(), encapsulating this logic and returning it as (say) a uint64_t?

Reply:
updated to `std::time_t`.

As for adding a helper method, I only found 3 other uses of `std::chrono::system_time::to_time_t` in the codebase, 2 of which used the value as `std::time_t` for further computations i.e. passing to APIs that expect `std::time_t`. So rather than `uint64_t`, I think it'd be helpful to return `std::time_t` which would still keep a bit of `<chrono>` logic in `time.h` but be consistent with some other APIs such as `server_.startTimeCurrentEpoch()` and `server_.startTimeFirstEpoch()` which return `std::time_t`. WDYT?

Reply:
ok, sounds good.